src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.OSX.cs

```diff
@@ -25,7 +25,33 @@ private static string GetProcPath(int processId)
             // Negative PIDs aren't valid
             ArgumentOutOfRangeException.ThrowIfNegative(pid);
 
-            ProcessInfo procInfo;
+            string? processName = null;
+
+            try
+            {
+                // Extract the process name from its path, because other alternatives such as
+                // reading proc_taskallinfo.pbsd.pbi_comm are limited in length
+                string processPath = GetProcPath(pid);
+                processName = Path.GetFileName(processPath);
+            }
+            catch
+            {
+                // Ignored
+            }
+
+            // Fallback to empty string if the process name could not be retrieved
+            processName ??= "";
```

Comment:
Can we fallback to pbi_comm if GetProcPath fails instead?

Reply:
Yes, we could fall back to pbi_comm if we have the proc_taskallinfo, and otherwise just fall back to the empty string again.

I'll work on these changes 👍